pkg/file/person.go

```diff
@@ -0,0 +1,231 @@
+	}
+
+	err := p.Payer.Validate()
+	if err != nil {
+		return err
+	}
+
+	for _, payee := range p.Payees {
+		err = payee.Validate()
+		if err != nil {
+			return err
+		}
+	}
+
+	err = p.EndPayer.Validate()
+	if err != nil {
+		return err
+	}
+
+	for _, state := range p.States {
+		err = state.Validate()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// SequenceNumber returns sequence number of the record
+func (p *paymentPerson) SequenceNumber() int {
+	if p.Payer == nil {
+		return 0
+	}
+	return p.Payer.SequenceNumber()
+}
+
+// SequenceNumber set sequence number of the record
+func (p *paymentPerson) SetSequenceNumber(int) {}
+
+// Parse attempts to parse with raw data.
+func (p *paymentPerson) Parse(buf []byte) (int, error) {
+	bufSize := len(buf)
+	readPtr := 0
+
+	if string(buf[readPtr]) != config.ARecordType || bufSize < config.RecordLength {
+		return readPtr, utils.ErrInvalidAscii
+	}
+
+	if p.Payer == nil {
+		p.Payer = records.NewARecord()
+	}
+	err := p.Payer.Parse(buf[readPtr : readPtr+config.RecordLength])
+	if err != nil {
+		return readPtr, err
+	}
+	readPtr += config.RecordLength
+
+	typeOfReturn := ""
+	if p.Payer != nil {
+		typeOfReturn = config.TypeOfReturns[p.Payer.(*records.ARecord).TypeOfReturn]
+	}
+
+	p.Payees = []records.Record{}
+	for string(buf[readPtr]) == config.BRecordType {
+		if bufSize < readPtr+config.RecordLength {
+			return readPtr, utils.ErrInvalidAscii
+		}
+
+		newPayee := records.NewBRecord(typeOfReturn)
+		if err = newPayee.Parse(buf[readPtr : readPtr+config.RecordLength]); err != nil {
+			return readPtr, err
+		}
+
+		readPtr += config.RecordLength
+		p.Payees = append(p.Payees, newPayee)
+	}
+
+	if string(buf[readPtr]) == config.CRecordType || bufSize < readPtr+config.RecordLength {
+		if p.EndPayer == nil {
+			p.EndPayer = records.NewCRecord()
+		}
+		err := p.EndPayer.Parse(buf[readPtr : readPtr+config.RecordLength])
+		if err != nil {
+			return readPtr, err
+		}
+		readPtr += config.RecordLength
+	}
+
+	p.States = []records.Record{}
+	for string(buf[readPtr]) == config.KRecordType {
+		if bufSize < readPtr+config.RecordLength {
+			return readPtr, utils.ErrInvalidAscii
+		}
+
+		newState := records.NewKRecord()
+		if err = newState.Parse(buf[readPtr : readPtr+config.RecordLength]); err != nil {
+			return readPtr, err
+		}
+
+		readPtr += config.RecordLength
+		p.States = append(p.States, newState)
+	}
+
+	return readPtr, nil
+}
+
+// UnmarshalJSON parses a JSON blob
+func (p *paymentPerson) UnmarshalJSON(data []byte) error {
+
+	dummy := make(map[string]interface{})
+	err := json.Unmarshal(data, &dummy)
+	if err != nil {
+		return err
+	}
+
+	for name, record := range dummy {
+		if name != "payer" {
+			continue
+		}
+		p.Payer = records.NewARecord()
+		err := readJsonWithRecord(p.Payer, record)
+		if err != nil {
+			return err
+		}
+	}
+
+	typeOfReturn := ""
+	if p.Payer != nil {
+		typeOfReturn = config.TypeOfReturns[p.Payer.(*records.ARecord).TypeOfReturn]
```

Comment:
Can this return records of a type besides `ARecord`?

Reply:
no, we should use ARecord from p.Payer only
there isn't type convert check for p.Payer.(*records.ARecord). it seems that the check is unmeaningful

Reply:
That's true. It worries me to have unchecked casts even if the `ARecord.Parse(..)` would only return that type. It calls out to `utils.ParseValue` which might get accidentally broken in the future. 

I added a type check on the cast in https://github.com/moov-io/irs/commit/85db345624afb0b66aaba7844ddcbb499ba69d8c